shared/subspace-data-retrieval/src/object_fetcher.rs

```diff
@@ -162,7 +175,9 @@ where
         Self {
             piece_getter,
             erasure_coding,
-            max_object_len: max_object_len.unwrap_or(usize::MAX),
+            max_object_len: max_object_len
+                .unwrap_or(usize::MAX)
+                .min(MAX_SUPPORTED_OBJECT_LENGTH),
```

Comment:
Why `.unwrap_or(usize::MAX)` and not `.unwrap_or(MAX_SUPPORTED_OBJECT_LENGTH)`?

Also I'd prefer making `max_object_len` non-optional, this is getting too confusing (it accepts the input, but actually limits the size internally).